gloo/transport/tcp/tls/openssl/dynamic_library.cc

```diff
@@ -0,0 +1,35 @@
+#include "dynamic_library.h"
+
+#include <stdexcept>
+#include <dlfcn.h>
+
+DynamicLibrary::DynamicLibrary(const char *name, const char *alt_name) : lib_name(name) {
+  handle = dlopen(name, RTLD_LOCAL | RTLD_NOW);
+  if (!handle) {
+    if (alt_name == nullptr) {
+      error = dlerror();
+    } else {
```

Comment:
Few comments here:
 - Feel free to get rid of `alt_name` as it is needed only for specific usecases in PyTorch
 - It's OK to throw runtime error from the constructor
 - If you want to preserve error message, please copy it into std::string, as message returned by dlerror() might get clobbered by subsequent `dlopen` call.

Reply:
addressed